environment.yml

```diff
@@ -1,11 +1,12 @@
-# Usage: conda env create -n myenvname -f environment.yml python=3.6
+# Usage: conda env create -n myenvname -f environment.yml
 ---
 name: aiida
 channels:
 - defaults
 - conda-forge
 - etetoolkit
 dependencies:
+- python~=3.7
```

Comment:
Does this mean that `aiida-core` can now only be installed through conda with python>=3.7? Can someone with python 3.5 still install through conda? Because we do support 3.5 and 3.6

Reply:
somebody using this environment.yml without changes will install aiida-core into an environment that has python>=3.7.

However, this would happen even if we were to remove this line (since - I believe - conda will choose the latest python version available in that case).

The only way to force (& test) build using other python versions would be to create copies of this file on travis that fix the minor version to 3.5, 3.6 and 3.7 and create an environment for each.

Reply:
Sorry for being a pain, but I am afraid I still don't fully understand. I thought you showed that installing `aiida-core` with conda will result in `python==3.6.10` being used, because apparently, various dependencies are not compatible with `python>=3.7`. Why then, are we here requiring `python~=3.7` when we know this should not work with conda _and_ as far as `aiida-core` is concerned, we support `python~=3.5`?

Reply:
Sure, no problem ;-)

>  I thought you showed that installing aiida-core with conda will result in python==3.6.10 being used, because apparently, various dependencies are not compatible with python>=3.7

Correct, but then I fixed that by making plumpy available under python 3.7 (otherwise the tests on this PR would fail).

Reply:
So it was really just `plumpy` that was the only dependency not compatible with python 3.7?

Reply:
Indeed - from conda's the error message I would have never guessed this.
Big thanks to @astamminger for figuring this one out.

Reply:
OK all clear now, thanks a lot to both!